src/sentry/integrations/discord/actions/metric_alert.py

```diff
@@ -51,13 +56,19 @@ def send_incident_alert_notification(
     )
 
     client = DiscordClient()
-    try:
-        client.send_message(channel, message)
-    except Exception as error:
-        logger.warning(
-            "discord.metric_alert.message_send_failure",
-            extra={"error": error, "incident_id": incident.id, "channel_id": channel},
-        )
-        return False
-    else:
+    with MessagingInteractionEvent(
+        interaction_type=MessagingInteractionType.SEND_INCIDENT_ALERT_NOTIFICATION,
+        spec=DiscordMessagingSpec(),
+    ).capture() as lifecycle:
+        try:
+            client.send_message(channel, message)
+        except Exception as error:
+            # TODO(iamrajjoshi): Update some of these failures to halts
+            lifecycle.record_failure(error)
+            # TODO(iamrajjoshi): Remove the logger after we audit lifecycle
```

Comment:
does auditing lifecycle mean we want more detail/cases into why this action will fail ?

Reply:
yeah once we are confident in our SLOs and our event lifecycle,  we can remove the logging since the lifecycle already handles logging and such for us.